classification/classification_network.py

```diff
@@ -246,109 +220,368 @@ def createClassifierRegion(network, classifierType, classifierParams):
+
+  # Create UP region, if enabled.
+  if networkConfiguration[UP_REGION_NAME]["enabled"]:
+    upParams = networkConfiguration[UP_REGION_NAME]["params"]
+    upRegion = _createUnionPoolerRegion(network, upParams)
+    # TODO: not sure about the UP region width params. This needs to be updated.
+    _validateRegionWidths(previousRegionWidth,
+                          upRegion.getSelf().cellsPerColumn)
+    _linkRegions(network,
+                 SENSOR_REGION_NAME,
+                 previousRegion,
+                 UP_REGION_NAME)
+    previousRegion = UP_REGION_NAME
+
+  # Create classifier region (always enabled)
+  classifierParams = networkConfiguration[CLASSIFIER_REGION_NAME]["params"]
+  classifierType = networkConfiguration[CLASSIFIER_REGION_NAME]["type"]
+  _createClassifierRegion(network,
+                          CLASSIFIER_REGION_NAME,
+                          classifierType,
+                          classifierParams)
+  # Link the classifier to previous region and sensor region - to send in 
+  # category labels.
+  network.link(previousRegion, CLASSIFIER_REGION_NAME, "UniformLink", "")
+  network.link(SENSOR_REGION_NAME,
+               CLASSIFIER_REGION_NAME,
+               "UniformLink",
+               "",
+               srcOutput="categoryOut",
+               destInput="categoryIn")
+
+  return network
 
-  sp = createSpatialPoolerRegion(network,
-                                 sensor.encoder.getWidth())
 
-  tm = createTemporalMemoryRegion(network)
 
-  classifier = createClassifierRegion(network,
-                                      classifierType,
-                                      classifierParams)
+def _enableRegionLearning(network,
+                          trainedRegionNames,
+                          regionName,
+                          recordNumber):
+  """
+  Enable learning for a specific region.
+  
+  @param network (Network) the network instance 
+  @param trainedRegionNames (list) regions that have been trained on the 
+  input data.
+  @param regionName (str) name of the current region  
+  @param recordNumber (int) value of the current record number 
+  """
 
-  validateRegions(sensor, sp, tm, classifier)
+  region = network.regions[regionName]
+  region.setParameter("learningMode", True)
+  phaseInfo = ("-> Training '%s'. RecordNumber=%s. Learning is ON for %s, "
+               "but OFF for the remaining regions." % (regionName,
+                                                       recordNumber,
+                                                       trainedRegionNames))
+  print phaseInfo
 
-  linkRegions(network)
 
+
+def _stopLearning(network, trainedRegionNames, recordNumber):
+  """
+  Disable learning for all trained regions.
+  
+  @param network (Network) the network instance 
+  @param trainedRegionNames (list) regions that have been trained on the 
+  input data.
+  @param recordNumber (int) value of the current record number 
```

Comment:
as explained above, I use this for printing out useful phase info. It' helpful to validate what region is learning / stops learning, at what point in the training set